src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -2134,137 +2799,196 @@ public enum RisingRivals implements LogicCardInfo {
+                  }
+                }
+                shuffleDeck()
+              }
+            }
           }
-          playRequirement{
+          onRemoveFromPlay{
+            actions.each { bg().gm().unregisterAction(it) }
           }
         };
       case SUNYSHORE_CITY_GYM_94:
-        return basicTrainer (this) {
+        return stadium (this) {
           text "This card stays in play when you play it. Discard this card if another Stadium card comes into play. If another card with the same name is in play, you can’t play this card.\nAny damage done by attacks from [L] Pokémon (both yours and your opponent’s) to the Defending Pokémon isn’t affected by Resistance. Each [L] Pokémon in play (both yours and your opponent’s) has no Weakness."
+          def eff1, eff2
           onPlay {
+            eff1 = getter (GET_WEAKNESSES) {h->
+              if(h.effect.target.types.contains(L)){
+                h.object = []
+              }
+            }
+            eff2 = delayed {
+              before APPLY_RESISTANCE, {
+                bg.dm().each {
+                  if (it.from.types.contains(L) && it.to.active) {
+                    prevent()
+                  }
+                }
+              }
+            }
           }
-          playRequirement{
+          onRemoveFromPlay{
+            eff1.unregister()
+            eff2.unregister()
           }
         };
       case TECHNICAL_MACHINE_G_95:
         return basicTrainer (this) {
           text "Attach this card to 1 of your Pokémon SP in play. That Pokémon may use this card’s attack instead of its own. When the Pokémon this card is attached to is no longer Pokémon SP, discard this card.\n[C][C][C] Damage Porter Choose 1 of your opponent’s Pokémon. This attack does 10 damage times the number of damage counters on the Pokémon this card is attached to that Pokémon. (Don’t apply Weakness and Resistance for Benched Pokémon.)"
-          onPlay {
+          onPlay {//TODO Implement technical machine groovy card type
           }
           playRequirement{
           }
         };
       case SP_RADAR_96:
-        return basicTrainer (this) {
+        return itemCard (this) {
           text "Choose a card from your hand and put it on top of your deck. Search your deck for a Pokémon SP, show it to your opponent, and put it into your hand. Shuffle your deck afterward. (If this is the only card in your hand, you can’t play this card.)"
           onPlay {
+            my.hand.getExcludedList(thisCard).select("Choose a card to put back into your deck").moveTo(hidden:true,my.deck)
+            def card = my.deck.search("Search your deck for a Pokémon SP",cardTypeFilter(POKEMON_SP))
+            if(card) {
+              card.showToOpponent("SP Radar: Selected Card").moveTo(my.hand)
+            }
+            shuffleDeck()
           }
           playRequirement{
+            assert my.hand.getExcludedList(thisCard) : "You have no other cards in your hand"
```

Comment:
Do you have any reference to ruling saying you can do this if you have no cards in your deck currently?